src/ir/dot/dot_generator.cc

```diff
@@ -144,14 +148,20 @@ std::string DotGeneratorHelper::GetDotNode(const Operation& op) {
                         absl::StrFormat(R"(<I%d>I%d)", node_num, node_num));
       });
   auto find_result = operation_results_.find(node_name);
+  const absl::btree_set<std::string>& results =
+      (find_result == operation_results_.end()) ? empty_results_set
+                                                : find_result->second;
```

Comment:
I am using a btree_set to get a predicable output. The alternative was to keep a `flat_hash_set` or `vector` and later sort it. However, shouldn't the asymptotic complexity be the same for both uses?

Reply:
The asymptotic complexity is the same, but often in C++, the performance penalty for using an online-sorting datastructure like a btree is worse than you might expect from complexity alone due to the less dense layout in memory (and thus additional cache misses) than something like a `std::vector` or an `absl::flat_hash_set` would provide. I usually try to nudge towards the dense structures at the time when a sorting structure is introduced. But since it's already in, and this isn't exactly a performance-critical piece of code, I think this is probably fine.